plugins/Disintegrator/Disintegrator.cpp

```diff
@@ -0,0 +1,279 @@
+		switch (type)
+		{
+			case 0:// Mono Noise
+			{
+				newInBufLoc[0] = fast_rand() / (float)FAST_RAND_MAX;
+
+				calcHighpassFilter(newInBufLoc[0], newInBufLoc[0], 0, lowCut, 0.707, sample_rate);
+				calcLowpassFilter(newInBufLoc[0], newInBufLoc[0], 0, highCut, 0.707, sample_rate);
+
+				newInBufLoc[0] = realfmod(m_inBufLoc - newInBufLoc[0] * amount, 200);
+				newInBufLoc[1] = newInBufLoc[0];
+
+				newInBufLocFrac[0] = fmod(newInBufLoc[0], 1);
+				newInBufLocFrac[1] = newInBufLocFrac[0];
+
+				break;
+			}
+			case 1:// Stereo Noise
+			{
+				newInBufLoc[0] = fast_rand() / (float)FAST_RAND_MAX;
+				newInBufLoc[1] = fast_rand() / (float)FAST_RAND_MAX;
+
+				calcHighpassFilter(newInBufLoc[0], newInBufLoc[0], 0, lowCut, 0.707, sample_rate);
+				calcHighpassFilter(newInBufLoc[1], newInBufLoc[1], 1, lowCut, 0.707, sample_rate);
+				calcLowpassFilter(newInBufLoc[0], newInBufLoc[0], 0, highCut, 0.707, sample_rate);
+				calcLowpassFilter(newInBufLoc[1], newInBufLoc[1], 1, highCut, 0.707, sample_rate);
+
+				newInBufLoc[0] = realfmod(m_inBufLoc - newInBufLoc[0] * amount, 200);
+				newInBufLoc[1] = realfmod(m_inBufLoc - newInBufLoc[1] * amount, 200);
+
+				newInBufLocFrac[0] = fmod(newInBufLoc[0], 1);
+				newInBufLocFrac[1] = fmod(newInBufLoc[1], 1);
+
+				break;
+			}
+			case 2:// Sine Wave
+			{
+				m_sineLoc = fmod(m_sineLoc + (freq / (float)sample_rate * F_2PI), F_2PI);
+
+				newInBufLoc[0] = (sin(m_sineLoc) + 1) * 0.5f;
+
+				newInBufLoc[0] = realfmod(m_inBufLoc - newInBufLoc[0] * amount, 200);
+				newInBufLoc[1] = newInBufLoc[0];
+
+				newInBufLocFrac[0] = fmod(newInBufLoc[0], 1);
+				newInBufLocFrac[1] = newInBufLocFrac[0];
+
+				break;
+			}
+		}
+
+		for (int b = 0; b < 2; ++b)
+		{
+			if (fmod(newInBufLoc[b], 1) == 0)
+			{
+				s[b] = m_inBuf[b][newInBufLoc[b]];
+			}
+			else
+			{
+				if (newInBufLoc[b] < 199)
+				{
+					s[b] = m_inBuf[b][floor(newInBufLoc[b])] * (1 - newInBufLocFrac[b]) + m_inBuf[b][ceil(newInBufLoc[b])] * newInBufLocFrac[b];
+				}
+				else
+				{
+					s[b] = m_inBuf[b][199] * (1 - newInBufLocFrac[b]) + m_inBuf[b][0] * newInBufLocFrac[b];
+				}
+			}
+		}
+
+		buf[f][0] = d * buf[f][0] + w * s[0];
+		buf[f][1] = d * buf[f][1] + w * s[1];
+	}
+
+	checkGate(outSum / frames);
+
+	return isRunning();
+}
+
+
+
+inline void DisintegratorEffect::calcLowpassFilter(sample_t &outSamp, sample_t inSamp, int which, float lpCutoff, float resonance, sample_rate_t Fs)
+{
+	if (m_prevLPCutoff[which] != lpCutoff)
+	{
+		m_prevLPCutoff[which] = lpCutoff;
+		const float w0 = D_2PI * lpCutoff / Fs;
+		const float tempCosW0 = cos(w0);
+		const float alpha = sin(w0) / 0.3535f;
+
+		m_LPa0 = 1 + alpha;
+		m_LPb0 = (1 - tempCosW0) * 0.5f;
+		m_LPb1 = 1 - tempCosW0;
+		m_LPa1 = -(2 * tempCosW0);
+		m_LPa2 = 1 - alpha;
+	}
+
+	m_filtLPY[which][0] = (m_LPb0 * (inSamp + m_filtLPX[which][2]) + m_LPb1 * m_filtLPX[which][1] - m_LPa1 * m_filtLPY[which][1] - m_LPa2 * m_filtLPY[which][2]) / m_LPa0;
```

Comment:
This line looks too long. Can you break this to 2 or 3 lines for better readability?